vendor/arm/mali/mali_profiler.cpp

```diff
@@ -47,13 +45,33 @@ struct MaliHWInfo
 	unsigned l2_slices;
 };
 
-MaliHWInfo get_mali_hw_info(const char *path)
+MaliHWInfo get_mali_hw_info(const MaliProfiler *profiler, const char *path)
 {
-	int fd = open(path, O_RDWR);        // NOLINT
+	struct ScopedFileHandle
+	{
+		int fd;
+		ScopedFileHandle(int fd) :
+		    fd(fd)
+		{}
+		~ScopedFileHandle()
+		{
+			if (fd >= 0)
+			{
+				close(fd);
+			}
+		}
+	};
+
+	MaliHWInfo hw_info;        // NOLINT
+	memset(&hw_info, 0, sizeof(hw_info));
+
+	int              fd = open(path, O_RDWR);        // NOLINT
+	ScopedFileHandle scoped_file_handle(fd);
```

Comment:
The `scoped_file_handle` here is unused, but is necessary for managing the lifetime of the file. Perhaps to make this relationship more explicit, it would be better to have the code after here use `scoped_file_handle.fd` instead of `fd`?

Reply:
Both suggestions have been implemented 👍